File: 1046-last-stone-weight/1046-last-stone-weight.java
class Solution {
    public int lastStoneWeight(int[] stones) {
        PriorityQueue<Integer> pq = new PriorityQueue<>(Collections.reverseOrder());
        for(int i = 0; i<stones.length; i++){
            pq.add(stones[i]);
        }
        while(pq.size()>=2){
            int f = pq.peek();
            pq.poll();
            int s = pq.peek();
            pq.poll();
            if(s!=f){
                pq.add(f-s);
            }
        }
        return pq.isEmpty() ? 0 : pq.peek();
    }
}

File: 1046-last-stone-weight/1046-last-stone-weight.cpp
class Solution {
public:
    int lastStoneWeight(vector<int>& stones) {
        priority_queue<int>pq(stones.begin(), stones.end());
        while(pq.size()>=2){
            int f = pq.top();
            pq.pop();
            int s = pq.top();
            pq.pop();
            if(s!=f){
                pq.push(f-s);
            }
        }
        return pq.empty() ? 0 : pq.top();
    }
};